spec/lib/api/2.0/catalogs-spec.js

```diff
@@ -0,0 +1,117 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+describe('Http.Api.Catalogs', function () {
+
+    var Promise;
+    var Errors;
+    var stubNeedByIdentifier;
+    var stubFind;
+
+    before('start HTTP server', function () {
+        this.timeout(5000);
+        return helper.startServer([]).then(function () {
+            Promise = helper.injector.get('Promise');
+            Errors = helper.injector.get('Errors');
+            var w = helper.injector.get('Services.Waterline');
+            stubNeedByIdentifier = sinon.stub(w.catalogs, "needByIdentifier");
+            stubFind = sinon.stub(w.catalogs, "find");
+        });
+    });
+
+    after('stop HTTP server', function () {
+        return helper.stopServer();
+    });
+
+    beforeEach("reset stubs", function() {
+        stubFind.reset();
+        stubNeedByIdentifier.reset();
+    });
+
+    describe("GET /catalogs", function() {
+
+        it("should a list of all catalogs", function() {
+
+            stubFind.returns(Promise.resolve([
+                {
+                    id: '123',
+                    node: '123',
+                    source: 'foo',
+                    data: { something: 'here' }
+                }
+            ]));
+
+            return helper.request().get('/api/2.0/catalogs')
+                .expect('Content-Type', /^application\/json/)
+                .expect(200)
+                .expect(function (res) {
+                    expect(res.body).to.be.an("Array").with.length(1);
+                    expect(res.body[0]).to.be.an("Object").with.property('id', "123");
+                });
+        });
+
+        it("should return an empty array if no catalogs exist", function() {
+
+            stubFind.returns(Promise.resolve([]));
```

Comment:
FYI because we use the `sinon-as-promised` library, you can also express this as: `stubFind.resolves([ ])`